File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        Set<Integer> lostSet= new HashSet<>();
        Set<Integer> reserveSet = new HashSet<>();
        
        for(int l : lost){
            lostSet.add(l);
        }
        
        for(int r : reserve){
            //여벌 체육복을 가져온 학생이 체육복을 도난
            if(lostSet.contains(r)){
                lostSet.remove(r);
            }else{
            reserveSet.add(r);
            }
        }
        
        List<Integer> sortedLost = new ArrayList<>(lostSet);
        Collections.sort(sortedLost);
        
        for(int s : sortedLost){
            if(reserveSet.contains(s-1)){
                reserveSet.remove(s-1);
            }else if(reserveSet.contains(s+1)){
                reserveSet.remove(s+1);
            }else{
                n--;
            }
        }
        
        return n;
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    lost = set(lost)
    reserve = set(reserve)
    
    inter = lost & reserve
    lost -= inter
    reserve -= inter
    
    for i in reserve:
        if i-1 in lost:
            lost.remove(i-1)
        elif i+1 in lost:
            lost.remove(i+1)
    return n-len(lost)